Upstream: No
Author: Rasmus Thomsen <oss@cogitri.dev>
Reason: We have a mismatch between the name of our target triplet and the
name of the ld for musl
diff --git a/lib/Driver/ToolChains/Linux.cpp b/lib/Driver/ToolChains/Linux.cpp
index bff1ab10..68ca89d9 100644
--- a/lib/Driver/ToolChains/Linux.cpp
+++ b/lib/Driver/ToolChains/Linux.cpp
@@ -434,6 +434,11 @@ std::string Linux::getDynamicLinker(const ArgList &Args) const {
       ArchName = "armeb";
       IsArm = true;
       break;
+    // FIXME: The Triplet name for x86 on Alpine Linux (i686) doesn't match
+    // up with the name for the ld (i386)!
+    case llvm::Triple::x86:
+      ArchName = "i386";
+      break;
     default:
       ArchName = Triple.getArchName().str();
     }
